newhelm/dependency_injection.py

```diff
@@ -11,28 +11,40 @@
 
 def inject_dependencies(
```

Comment:
I think the modifications to this method make the most common use case (setting dependencies) unnecessarily complicated to support a rare use case (check dependencies).

I think we should keep the signature of this method, and instead create a peer that is something like `list_dependency_usage` which returns:

* All dependencies that were successfully read.
* All dependencies that were required but not provided.
* (Maybe) all dependencies that were optional but not provided.

Sorry for potentially misleading you with my previous comment.